# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerRef.cpp
# Commit: 13780f5cb2f5
# Full Hash: 13780f5cb2f5d698f35a23f2fafa1593ef7affe9
# Author: Joshua Marshall <jmarshall@mozilla.com>
# Date: 2024-01-28 21:22:06
# Regressor Bug: 1805613
# File Overlap Count: 1
# Description:
#   Bug 1805613 - Add a mutable status to WorkerRefs for logging. r=dom-worker-reviewers,asuth
#   
#   In DEBUG mode, add a settable string to WorkerRefs.  Record what we're doing in
#   the XHR request worker, and log the status if we seem to be hanging (see
#   WorkerPrivate::DumpCrashInformation).
# ==============================================================================

diff -r 1b609eb33a5f -r 13780f5cb2f5 dom/workers/WorkerRef.cpp
--- a/dom/workers/WorkerRef.cpp	Sun Jan 28 13:43:54 2024 +0100
+++ b/dom/workers/WorkerRef.cpp	Sun Jan 28 14:28:44 2024 +0000
@@ -47,7 +47,11 @@
 
 WorkerRef::WorkerRef(WorkerPrivate* aWorkerPrivate, const char* aName,
                      bool aIsPreventingShutdown)
-    : mWorkerPrivate(aWorkerPrivate),
+    :
+#ifdef DEBUG
+      mDebugMutex("WorkerRef"),
+#endif
+      mWorkerPrivate(aWorkerPrivate),
       mName(aName),
       mIsPreventingShutdown(aIsPreventingShutdown),
       mHolding(false) {